model/task/db.go

```diff
@@ -292,11 +298,16 @@ var (
 	AddAnnotations = []bson.M{
 		{
 			"$facet": bson.M{
```

Comment:
I don't think this is the correct approach. If the version was created after the annotations cache key was added we shouldn't even try to do this `$facet`. `$facets` have a lot of [limitations](https://www.mongodb.com/docs/manual/reference/operator/aggregation/facet/#considerations) and should generally be avoided. This was originally added because we wanted to reduce the amount of tasks we perform a `$lookup` for. 
After this is deployed we will hit the [100 MB aggregation limit](https://arc.net/l/quote/fcvemzqz) for facets on every single query we perform for `mongodb-mongo-master` project and it will [spill over to](https://github.com/evergreen-ci/evergreen/blob/c5495d494db06a2fb5a0422194dddd1df63ccb0f/model/task/db.go#L2431) the [`AddAnnotationsSlowLookup`](https://github.com/evergreen-ci/evergreen/blob/c5495d494db06a2fb5a0422194dddd1df63ccb0f/model/task/db.go#L362) and it will eliminate all performance gains this pr introduces. (I think it would actually become dramatically slower)